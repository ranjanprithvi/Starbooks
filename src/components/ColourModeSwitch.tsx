import { HStack, Switch, Text, useColorMode } from "@chakra-ui/react";

const ColourModeSwitch = ({ ...rest }) => {
    const { toggleColorMode, colorMode } = useColorMode();

    return (
        <HStack {...rest}>
            <Switch
                isChecked={colorMode === "dark"}
                onChange={toggleColorMode}
                colorScheme="teal"
            />

            <Text>Dark Mode</Text>
        </HStack>
    );
};

export default ColourModeSwitch;
