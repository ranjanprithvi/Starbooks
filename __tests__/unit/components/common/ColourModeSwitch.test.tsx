import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import ColourModeSwitch from "../../../../src/components/common/ColourModeSwitch";
import React from "react";
import { fireEvent, render } from "@testing-library/react";

const mToggleColorMode = jest.fn();
// const mUseColourMode = jest.fn().mockReturnValue();
jest.mock("@chakra-ui/react", () => {
    const originalModule = jest.requireActual("@chakra-ui/react");

    return {
        __esModule: true,
        ...originalModule,
        useColorMode: () => {
            return { colorMode: "light", toggleColorMode: mToggleColorMode };
        },
    };
});

describe("ColourModeSwitch", () => {
    it("renders correctly", () => {
        const tree = renderer.create(<ColourModeSwitch />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls toggleColourMode when clicked", () => {
        const { getByTestId } = render(<ColourModeSwitch />);

        fireEvent.click(getByTestId("toggler"));
        expect(mToggleColorMode).toHaveBeenCalled();
    });
});
