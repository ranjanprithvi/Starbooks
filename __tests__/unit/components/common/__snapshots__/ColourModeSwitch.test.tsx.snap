// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColourModeSwitch renders correctly 1`] = `
<div
  className="chakra-stack css-1igwmid"
>
  <label
    className="chakra-switch css-16pgy8f"
    data-testid="toggler"
    onClick={[Function]}
  >
    <input
      checked={false}
      className="chakra-switch__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      style={
        {
          "border": "0",
          "clip": "rect(0, 0, 0, 0)",
          "height": "1px",
          "margin": "-1px",
          "overflow": "hidden",
          "padding": "0",
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": "1px",
        }
      }
      type="checkbox"
    />
    <span
      aria-hidden={true}
      className="chakra-switch__track css-14qxnv8"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
    >
      <span
        className="chakra-switch__thumb css-0"
      />
    </span>
  </label>
  <p
    className="chakra-text css-0"
  >
    Dark Mode
  </p>
</div>
`;
